Patches/CustomDataLoader/CreateCardClonesPostfix.cs

```diff
@@ -35,29 +35,92 @@ static void LoadCharacterData(Dictionary<string, SubClassData> ____SubClass)
                 {
                     continue;
                 }
-                if (classes.Contains(newCharacter.SubClassName))
+
+                if (newCharacter.cardCounts?.Length > 0 && newCharacter.cardIds?.Length > 0)
                 {
-                    if (newCharacter.cardCounts != null && newCharacter.cardIds != null)
+                    Plugin.Logger.LogInfo($"Setting cards for {newCharacter.SubClassName}");
+                    List<HeroCards> heroCardsList = new List<HeroCards>();
+                    for (int i = 0; i < newCharacter.cardIds.Length; i++)
                     {
-                        Plugin.Logger.LogInfo(newCharacter.SubClassName);
-                        List<HeroCards> heroCardsList = new List<HeroCards>();
-                        for (int i = 0; i < newCharacter.cardIds.Length; i++)
+                        HeroCards heroCards = new HeroCards();
+                        if (Globals.Instance.GetCardData(newCharacter.cardIds[i]) != null)
                         {
-                            HeroCards heroCards = new HeroCards();
-                            if (Globals.Instance.GetCardData(newCharacter.cardIds[i]) != null)
-                            {
-                                heroCards.Card = Globals.Instance.GetCardData(newCharacter.cardIds[i]);
-                                heroCards.UnitsInDeck = newCharacter.cardCounts[i];
-                                heroCardsList.Add(heroCards);
-                                ____SubClass[newCharacter.SubClassName].Cards.AddToArray(heroCards);
-                                Plugin.Logger.LogInfo($"Added card {newCharacter.cardIds[i]} with quantity {newCharacter.cardCounts[i]} to {newCharacter.SubClassName}");
-                            }
+                            heroCards.Card = Globals.Instance.GetCardData(newCharacter.cardIds[i]);
+                            heroCards.UnitsInDeck = newCharacter.cardCounts[i];
+                            heroCardsList.Add(heroCards);
+                            Plugin.Logger.LogInfo($"Added card {newCharacter.cardIds[i]} with quantity {newCharacter.cardCounts[i]} to {newCharacter.SubClassName}");
                         }
+                    }
+                    ____SubClass[newCharacter.SubClassName].Cards = heroCardsList.ToArray();

```

Comment:
maybe store `____SubClass[newCharacter.SubClassName]` into a variable so you don't have to access it this way all the time?